spec/factories/courses.rb

```diff
@@ -87,18 +87,28 @@
           course.level = "further_education"
         end
       end
+
+      if course.subjects.any?
+        generate_course_title_service = Courses::GenerateCourseTitleService.new
+        course.name = generate_course_title_service.execute(course: course)
+      end
     end
 
     after(:create) do |course, evaluator|
       # This is important to retain the relationship behaviour between
       # course and it's enrichment
-
       if evaluator.age.present?
         course.created_at = evaluator.age
         course.updated_at = evaluator.age
         course.changed_at = evaluator.age
       end
 
+      if course.subjects.any?
+        generate_course_title_service = Courses::GenerateCourseTitleService.new
+        course.name = generate_course_title_service.execute(course: course)
+        course.save
```

Comment:
no need to save, this is factory for create

Reply:
done